### PR TITLE
Support for singlelist provider

### PR DESCRIPTION
**Pull request by jurgenheine**
- Added support for single list 
- Refactor some code
- Changed folder and filename handling
- Changed m3u list to utf-8 encoding (otherwise I had strange characters)
- Added url part matching for vod type ( better and faster matching in my case)

I could not test with Apolo list or other multi list providers
The last 2 changes can possible interfere

### CODE REVIEW

File: .github/workflows/docker-image-local.yml
Comment (hooray4me) [inline .github/workflows/docker-image-local.yml:1]: I am using dockerhub for my container registry. I hadn't gotten around to cleaning up my explicit references.

Comment (jurgenheine) [reply .github/workflows/docker-image-local.yml:1]: You can remove the workflow, I use it to create a GitHub docker file for testing. I didn't look how to create a PR without it.

Comment (hooray4me) [reply .github/workflows/docker-image-local.yml:1]: I'm still looking thru this... I think you'd have to remove from your working directory, then push that commit to this PR? I'm not sure I can just remove the workflow.

File: tools.py [hunk 1]
```diff
@@ -245,6 +261,7 @@ def parseEpisode(title):
       episodenumber = episodeMatch(title)
       showtitle = stripSxxExx(title)
     return [showtitle, episodetitle, seasonnumber, episodenumber, language]
+  
 def compare_and_update(dcmp):
```
Comment (hooray4me) [inline tools.py:265]: I had updated the logic here in my script repo to also clean up old or changed files from the provider
```bash
def compare_and_update(dcmp):
    for name in dcmp.diff_files:
        print("STREAM CHANGE -  %s - UPDATING" % (name))
        if os.path.isdir(dcmp.left+"/"+name):
            shutil.copytree(dcmp.left+"/"+name, dcmp.right+"/"+name, dirs_exist_ok=True)
        elif os.path.isfile(dcmp.left+"/"+name):
            shutil.copy2(dcmp.left+"/"+name, dcmp.right+"/"+name)
    for name in dcmp.left_only:
        if os.path.isdir(dcmp.left+"/"+name):
            print("NEW STREAM DIRECTORY - %s - CREATING" % (name))
            shutil.copytree(dcmp.left+"/"+name, dcmp.right+"/"+name, dirs_exist_ok=True)
        elif os.path.isfile(dcmp.left+"/"+name):
            print("NEW STREAM FILE - %s - CREATING" % (name))
            shutil.copy2(dcmp.left+"/"+name, dcmp.right+"/"+name)
    for name in dcmp.right_only:
        if os.path.isdir(dcmp.right+"/"+name):
          print("directory NO LONGER EXISTS - %s - DELETING" % (name))
          shutil.rmtree(dcmp.right+"/"+name)
        if os.path.isfile(dcmp.right+"/"+name):
          print("file NO LONGER EXISTS - %s - DELETING" % (name))
          os.remove(dcmp.right+"/"+name)
    for sub_dcmp in dcmp.subdirs.values():
        compare_and_update(sub_dcmp)
```

File: .github/workflows/docker-image-local.yml
Comment (jurgenheine) [reply .github/workflows/docker-image-local.yml:1]: You can delete the file, but I will try to do a PR without this file

File: tools.py (hunk 1 above)
Comment (jurgenheine) [reply tools.py:265]: It has to be merged then, I did a merge before I created the PR so your last changes where in my repo also. I will look into into it, they probably crossed.

**Pull request closed without merging**